### ISSUE TITLE
Option to make IDE0028 not apply if type isn't visible elsewhere on the line

### ISSUE THREAD

**Issue by ThioJoe** [dotnet/roslyn-analyzers#7515]
### Analyzer

**Diagnostic ID**: [IDE0028](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0028)

### Describe the improvement

I'd like an option to not show the IDE0028 suggestion for existing variables, because it would typically result in the type not showing on the line at all, and I personally would rather the type be clearly apparent in such cases.

### For Example:

On a line like this:
```cs
_someProperty = new List<string>();
```
Where it suggests simplifying to this:
```cs
_someProperty = [];
```

However if the variable or property was declared far away in the code, I find it more clear to leave it with the explicit type showing.

As opposed to a line like this where the type is still visible on the left so I do indeed want to simplify it on the right.

```
List<string> someOtherVariable = [];
```

In other words I like the type to always be showing at least once on the line.
